### PR TITLE
Add Serverside capability

### PR DESCRIPTION
**Pull request by brandonsturgeon**

### CODE REVIEW

File: lua/cfc_http_restrictions/list_manager.lua [hunk 1]
```diff
@@ -83,42 +84,44 @@ function CFCHTTP.isHTMLAllowed( html )
 end
 
 
--- file based config functions
-function CFCHTTP.allowAddress( addr )
-    if CFCHTTP.config.addresses[addr] ~= nil and CFCHTTP.config.addresses[addr].permanent then
-        notification.AddLegacy( "You cant change this address", NOTIFY_ERROR, 5 )
-        return false
-    end
+if CLIENT then
```
Comment (plally) [inline lua/cfc_http_restrictions/list_manager.lua:87]: would be fine to define these on the server too they might be usefull if we ever implement a ui for editing server configs

File: lua/cfc_http_restrictions/config_loader.lua [hunk 2]
```diff
@@ -44,21 +62,31 @@ function CFCHTTP.mergeConfigs( old, new )
...
 
-    notification.AddLegacy( "Saved http whitelist", NOTIFY_GENERIC, 5 )
+        cb( config )
+    end )
 end
 
-function CFCHTTP.readFileConfig()
-    local fileData = file.Read( "cfc_cl_http_whitelist_config.json" )
-    if not fileData then return end
+if CLIENT then
```
Comment (plally) [inline lua/cfc_http_restrictions/config_loader.lua:81]: same here I dont think it would hurt to just run these on server

Comment (brandonsturgeon) [reply lua/cfc_http_restrictions/config_loader.lua]: Done ✔️

File: lua/cfc_http_restrictions/list_manager.lua (hunk 1 above)
Comment (brandonsturgeon) [reply lua/cfc_http_restrictions/list_manager.lua]: Done ✔️

File: lua/cfc_http_restrictions/config_loader.lua [hunk 3]
```diff
@@ -1,24 +1,42 @@
+AddCSLuaFile()
+
 CFCHTTP = CFCHTTP or {}
-CFCHTTP.config = include( "default_config.lua" )
+CFCHTTP.config = include( "cfc_http_restrictions/default_config.lua" )
+
+if SERVER then
+    CFCHTTP.svEnabled = CreateConVar( "cfc_http_restrictions_enabled", 0, FCVAR_ARCHIVE + FCVAR_PROTECTED, "Should the HTTP restrictions be enabled?", 0, 1 )
```
Comment (plally) [inline lua/cfc_http_restrictions/config_loader.lua:7]: could you document this in the README

**Review by plally — APPROVED** [review]

File: lua/cfc_http_restrictions/config_loader.lua (hunk 3 above)
Comment (plally) [inline lua/cfc_http_restrictions/config_loader.lua:7]: could you document this in the README and change the description/name to reflect that this only disables/enables it serverside

Comment (brandonsturgeon) [reply lua/cfc_http_restrictions/config_loader.lua:7]: Yeah I'll update the README to make it clear that it filters either realm, but SV is disabled by default

Comment (plally) [reply lua/cfc_http_restrictions/config_loader.lua:7]: I meant change the name to be `cfc_http_sv_restrictions_enabled` or something so theres no confusion with it affecting clientside restrictions

Comment (brandonsturgeon) [reply lua/cfc_http_restrictions/config_loader.lua:7]: ah i see, yes I'll do both

Comment (brandonsturgeon) [reply lua/cfc_http_restrictions/config_loader.lua]: New README: https://github.com/CFC-Servers/cfc_cl_http_whitelist/blob/522ac21557cc701cff980e86298d8016d9c8f30d/README.md

**Review by plally — APPROVED** [review]

**Comment by plally** [comment]
i think we should change this so configs/ is shared and loaded on both realms, and add a new configs/client for client only configs.

**Comment by brandonsturgeon** [comment]
Closing in favor of https://github.com/CFC-Servers/cfc_cl_http_whitelist/pull/57

**Pull request closed without merging**